### PR TITLE
Use RpcUrl from provider 

### PR DESCRIPTION
**Pull request by joepegler**
# Summary

Takes rpcUrl from provider assuming it hasn't been explicitly set

Related Issue: SMA-549

## Change Type

- [ ] Bug Fix
- [ ] Refactor
- [ ] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [x] Other

# Checklist

- [x] My code follows this project's style guidelines
- [x] I've reviewed my own code
- [x] I've added comments for any hard-to-understand areas
- [x] I've updated the documentation if necessary
- [x] My changes generate no new warnings
- [x] I've added tests that prove my fix is effective or my feature works
- [x] All unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: packages/account/tests/account.spec.ts [hunk 1]
```diff
@@ -40,6 +42,78 @@ describe("Account Tests", () => {
...
+    } = ganache;
+
+    const newRpcUrl = "http://localhost:8545";
+    const defaultRpcUrl = viemChain.rpcUrls.public.http[0]; //http://127.0.0.1:8545"
+
+    const ethersProvider = new JsonRpcProvider(newRpcUrl);
+    const ethersSignerWithNewRpcUrl = new Wallet(privateKey, ethersProvider);
+
+    const accountOne = privateKeyToAccount(privateKey);
+    const walletClientWithNewRpcUrl = createWalletClient({
```
Comment (livingrockrises) [inline packages/account/tests/account.spec.ts:60]: what is the difference between viem's createPublicClient and createWalletClient?

**Review by livingrockrises — APPROVED** [review]
LGTM

**Comment by livingrockrises** [comment]
resolve the conflicts

File: packages/account/tests/account.spec.ts (hunk 1 above)
Comment (joepegler) [reply packages/account/tests/account.spec.ts:60]: They just do different things. They both come from viem. In ethers terms it's almost the same difference as a provider and a signer.

**Pull request merged**